syntax = "proto3";

import "precompiled/value.proto";

package precompiled;

message SymbolInformation {
  QualifiedType qualified_type = 1;
  bytes value = 2;
  bool visible = 3;
}

message SymbolInformationList {
  repeated SymbolInformation symbol = 1;
}

message ModuleProto {
  // An identifier for this module unique amongst all modules being linked
  // together.
  string identifier = 1;

  // All functions available from this module, regardless of their visibility or
  // whether they are bound to names.
  repeated Function function = 2;

  map<string, SymbolInformationList> symbols = 3;

  TypeSystem type_system = 4;
}
